openedx/core/djangoapps/course_apps/api.py

```diff
@@ -0,0 +1,63 @@
+"""
+Python APIs for Course Apps.
+"""
+from django.contrib.auth import get_user_model
+from opaque_keys.edx.keys import CourseKey
+
+from .models import CourseAppStatus
+from .plugin import CourseAppsPluginManager
+
+
+User = get_user_model()
+
+
+def is_course_app_enabled(course_key: CourseKey, app_id: str) -> bool:
+    """
+    Return if the app with the specified `app_id` is enabled for the
+    specified course.
+
+    Args:
+        course_key (CourseKey): Course key for course
+        app_id (str): The app id for a course app
+
+    Returns:
+        True or False depending on if the course app is enabled or not.
+    """
+    course_app = CourseAppsPluginManager.get_plugin(app_id)
+    is_enabled = course_app.default_enabled
+    # For existing apps that doesn't support the new mechanism of enabling/disabling
+    # we delegate to the app to know whether it's enabled or disabled.
+    obj, _ = CourseAppStatus.objects.get_or_create(
+        course_key=course_key, app_id=app_id, defaults={"enabled": is_enabled}
+    )
+    if hasattr(course_app, "is_enabled"):
```

Comment:
Every app should implement its own `is_enabled` method.

Reply:
I'd call out that as "implementation details being exposed to the platform" with regards to "more control". 

The interface should be simple and do not expose unnecessary properties. I am saying this because a method can encapsulate the whole logic inside. 

```
Class App:

    def is_enabled_for_course(self, course, user):
        is_installed = check_if_installed()
        if not is_installed:
            return False
        is_available = check_is_available()
        if not is_available:
            return False
        is_enabled = check_is_enabled()
       return is_enabled
                
```
as a user of this class, I do not need to worry to check if it is installed first on or, or this object has "xyz" method on it or not. 

I think we should simplify the code coupling here.